CMSIS/DSP/Include/arm_math_types.h

```diff
@@ -161,6 +165,12 @@ extern "C"
   #define LOW_OPTIMIZATION_EXIT
   #define IAR_ONLY_LOW_OPTIMIZATION_ENTER
   #define IAR_ONLY_LOW_OPTIMIZATION_EXIT
+  
+#elif defined ( __APPLE_CC__ )
```

Comment:
Is \_\_GNUC\_\_ implied by \_\_APPLE\_CC\_\_ ?
Because at beginning and end of the file there are lots of compiler tests and there is no tests for \_\_APPLE\_CC\_\_.

The tests are beginning of file are ending with #error Unknown compiler
So if you can build it means the tests are passing and one of the symbol (\_\_GNUC\_\_ ?) must be defined ?

Reply:
Indeed, __GNUC__ is implied by __APPLE_CC__
However the attribute __attribute__(( optimize("-O1") )) used in case of __GNUC__  is a little problematic with Xcode:
"Unknown attribute 'optimize'"